nifi-nar-bundles/nifi-redis-bundle/nifi-redis-extensions/src/main/java/org/apache/nifi/redis/service/RedisDistributedMapCacheClientService.java

```diff
@@ -332,43 +127,4 @@ public <K, V> boolean replace(final AtomicCacheEntry<K, V, byte[]> entry, final
         });
     }
 
-    // ----------------- END Methods from AtomicDistributedMapCacheClient ------------------------
-
-    private <K, V> Tuple<byte[],byte[]> serialize(final K key, final V value, final Serializer<K> keySerializer, final Serializer<V> valueSerializer) throws IOException {
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
-
-        keySerializer.serialize(key, out);
-        final byte[] k = out.toByteArray();
-
-        out.reset();
-
-        valueSerializer.serialize(value, out);
-        final byte[] v = out.toByteArray();
-
-        return new Tuple<>(k, v);
-    }
-
-    private <K> byte[] serialize(final K key, final Serializer<K> keySerializer) throws IOException {
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
-
-        keySerializer.serialize(key, out);
-        return out.toByteArray();
-    }
-
-    private <T> T withConnection(final RedisAction<T> action) throws IOException {
```

Comment:
The connection handling looks the same to me. Am I missing something? I am not seeing where one use is transactional, but the other isn't.

Reply:
The transactional code is in the methods from `AtomicDistributedMapCache` that are implemented in `RedisDistributedMapCacheClientService`. Since we can't rename the existing service (it would be better to name that one `AtomicRedisDistributedMapCacheClientService`), we can leave that one alone and move the methods that are from only `DistributedMapCacheClientService` down to `SimpleRedisDistributedMapCacheClientService`. This if a processor declares that it only needs `DistributedMapCacheClient` (not atomic), then it can use the Simple Redis one even when Redis is clustered.